doc/network.md

```diff
@@ -0,0 +1,246 @@
+| size   | [VarI32] | The size of the array.                      |
+| 0      | A        | The first element of the array if present.  |
+| 1      | A        | The second element of the array if present. |
+| ...    | A        | An element of the array if present.         |
+| N-1    | A        | The N-1 element of the array if present.    |
+
+### Tuple `(A, B, ...)`
+
+A tuple is a type which holds the content of multiple types specified in its name.
+The tuple type is very abstract and acts as many fields as the tuple should hold.
+
+Size: size of `A` + size of `B` + ...
+
+| Fields | Type | Description                    |
+|:------:|:----:|:-------------------------------|
+| first  | `A`  | The first value of the tuple.  |
+| second | `B`  | The second value of the tuple. |
+| ...    | ...  | Another value of the tuple.    |
+
+### Optional `X?`
+
+An optional is a value which may or may not be present.
+
+Size: 0 if non-present, else size of `X`.
+
+## Packets
+
+### Workspace-related Packets
+
+The server controls most of the requests sent by a client for workspace-related packets as it requires some permissions.
+Those packets are most likely sent only on build servers.
+
+#### `plasmid:workspace/enter`
+
+Packet sent when a player enters a workspace editor. The client should render the bounds of the map if present,
+and the contained regions, points, etc.  
+A `plasmid:workspace/regions` packet should be sent with this packet if the workspace contains regions.
+
+Direction: `C<-S`
+
+| Fields    | Type       | Description                                                            |
+|:---------:|:----------:|:-----------------------------------------------------------------------|
+| workspace | Identifier | The identifier of the entered workspace.                               |
+| bounds    | [Bounds]   | The bounds of the map, may be `[[0, 0, 0], [0, 0, 0]]` if not present. |
+| world     | Identifier | The world in which the workspace is present.                           |
+| data      | NBT Tag    | Arbitrary data assigned to the map template.                           |
+
+#### `plasmid:workspace/new`
+
+Sent by the client to request the server to create a new workspace. 
+
+Direction: `C->S`
+
+| Fields    | Type           | Description                                                            |
+|:---------:|:--------------:|:-----------------------------------------------------------------------|
+| workspace | Identifier     | The identifier of the new workspace.                                   |
+| bounds    | [Bounds]       | The bounds of the map, may be `[[0, 0, 0], [0, 0, 0]]` if not present. |
+| generator | string (32767) | The generator string for the map, may be empty if unspecified.         |
+| has_data  | bool           | `true` if contains a data NBT tag, else `false`.                       |
+| data      | (NBT Tag)?     | Arbitrary data assigned to the map template.                           |
```

Comment:
The optional data tag like this doesn't seem to be used anywhere else- may be worth just forcing to send an empty data tag instead of sending none.